src/dotnet/CarbonAware.DataSources.WattTime.Tests/WattTimeDataSourceTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using CarbonAware.Tools.WattTimeClient;
+using CarbonAware.Tools.WattTimeClient.Model;
+using Castle.Core.Logging;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarbonAware.DataSources.WattTime.Tests;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+public class WattTimeDataSourceTests
+{
+    private Mock<ILogger<WattTimeDataSource>> Logger { get; set; }
+
+    private Mock<IWattTimeClient> WattTimeClient { get; set; }
+
+    private ActivitySource ActivitySource { get; set; }
+
+    private WattTimeDataSource DataSource { get; set; }
+
+    private List<Forecast> Forcasts { get; set; }
+
+    private Mock<IRegionConverter> RegionConverter { get; set; }
+
+    private BalancingAuthority BalancingAuthority { get; set; }
+
+    private DateTimeOffset StartDate { get; set; }
+
+    private DateTimeOffset EndDate { get; set; }
+
+    private string Region { get; set; }
+
+    [SetUp]
+    public void Setup()
```

Comment:
Stylistically, this _feels_ wrong to declare test battery information as a bunch of magic strings and data inside the test object. Down the road, I'm not sure which of these variables I could change and still have the tests pass. How do I know that "market" needs to be "market" or that version is "1"?

Reply:
Moved this into the individual tests and removed some of the extra stuff.